woocommerce/woocommerce
# Code analyzer: find multiple hooks in single patch, add tests

### Changes proposed in this Pull Request:

@jonathansadowski correctly observed that the code analyzer could not pick up multiple hook changes in a single file patch. 

This PR enhances that code to pick up multiple hooks and also adds jest tests covering a real WooCommerce diff.



### How to test the changes in this Pull Request:

1. These tests are not run in CI yet (no rush to add in this PR I think) so checkout this branch, and run the tests from the `tools/code-analyzer` directory: `pnpm test`
2. You can see the corrected analyzer output for this real diff as well: `pnpm analyzer lint "release/6.8" "6.8.0" -b release/6.7` You should see all 6 hook/filter changes



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
